### ISSUE TITLE
vcpkg create doesn't work.

### ISSUE THREAD

**Issue by ChanthMiao** [microsoft/vcpkg#18406]
**Describe the bug**
`vcpkg create` doesn't work. It throws sha512 error.

**Environment**
- OS: windows 10 v10.0.19042.0
- shell: powershell Core 7.1.3
- vcpkg: 2021-05-05-9f849c4c43e50d1b16186ae76681c27b0c1be9d9

**To Reproduce**
1. `.\vcpkg.exe create everything https://github.com/ChanthMiao/Everything-SDK/archive/refs/tags/1.0.0.zip Everything-SDK-1.0.0.zip`
2. See error

**Expected behavior**
Generated portfile successifully.

**Failure logs** 
```powershell
CMake Error at scripts/cmake/vcpkg_download_distfile.cmake:88 (message):
  Invalid SHA512: .  If you do not know the file's SHA512, set this to "0".
Call Stack (most recent call first):
  scripts/ports.cmake:172 (vcpkg_download_distfile)
```

**Additional context**
The #2426 helps nothing.
I also tried `.\vcpkg create zlib2 http://zlib.net/zlib-1.2.11.tar.gz zlib-1.2.11.zip`, with the same error message.

**Comment by JackBoosY** [comment]
Confirmed.

**Comment by JackBoosY** [comment]
Fixed by #18459.

**Comment by ChanthMiao** [comment]
Solved.
